Training24/intlist.c

```diff
@@ -0,0 +1,179 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// intlist.c
+// Program on Linked list functions.
+// ------------------------------------------------------------------------------------------------
+#include <malloc.h>
+#include "intlist.h"
+
+int mlastError = SUCCESS;  // Global variable to store the last error code
+
+IntList* Create () {
+   IntList* newList = (IntList*)malloc (sizeof (IntList));
+   if (newList == NULL) {
+      mlastError = MEMORY_ALLOCATION_FAILED;
+      return NULL;
+   }
+   newList->head = NULL; ; // Intialilze head pointer as null
+   mlastError = SUCCESS;
+   return newList;
+}
+
+void Delete (IntList* list) {
+   if (list == NULL) {
+      mlastError = NULL_POINTER;

```

Comment:
This error message infers that the pointer is null. But here the check is for null list. Change the message accordingly.